rust/operator-binary/src/catalog/hive.rs

```diff
@@ -42,6 +45,29 @@ impl ToCatalogConfig for HiveConnector {
         // See https://trino.io/docs/current/connector/hive-security.html
         config.add_property("hive.security", "allow-all");
 
+        Ok(config)
+    }
+}
+
+#[async_trait]
+impl ToCatalogConfig for HiveAndIcebergCommonAttributes {
+    async fn to_catalog_config(
+        &self,
+        catalog_name: &str,
+        catalog_namespace: Option<String>,
+        client: &Client,
+    ) -> Result<CatalogConfig, FromTrinoCatalogError> {
+        let mut config = CatalogConfig::new(
+            catalog_name.to_string(),
+            "must be overwritten to hive or iceberg",
```

Comment:
This is very specific (see other comments to split this up).

Reply:
Problem here is that i have to stick to using Traits (ToCatalogConfig in this case) and therefore can't add a new parameter `connector_name`. I can't add random fns to HiveAndIcebergCommonAttributes.
In theory i could add a new Trait e.g. ToCatalogConfigWithConnectorName and implement that on HiveAndIcebergCommonAttributes. But i though that makes things more complicated than this here